### ISSUE TITLE
does this work with threads?

### ISSUE THREAD

**Issue by AlJohri** [reclosedev/requests-cache#118]
I tried using this library with `concurrent.futures.ThreadPoolExecutor` to fill up the cache like so:

```python
import requests
import requests_cache
from requests.adapters import HTTPAdapter
adapter = HTTPAdapter(max_retries=5)
session = requests_cache.core.CachedSession()
session.mount('http://', adapter)

from concurrent.futures import ThreadPoolExecutor, as_completed

urls = [....]

with ThreadPoolExecutor(max_workers=7) as executor:
    future_to_url = {executor.submit(session.get, url): url for url in urls}
    for i, future in enumerate(as_completed(future_to_url)):
        url = future_to_url[future]
        response = future.result()
```

It seemed to be working pretty slowly- I wasn't sure exactly what was going on. Do you know if this is supported?

**Comment by reclosedev** [comment]
Yes, it should work fine with threads. It allows only one thread to access DB at a time using locks.
As for speed issue, you can try to pass ``fast_save`` parameter to session (dbdict storage option)
https://requests-cache.readthedocs.io/en/latest/api.html#requests_cache.backends.sqlite.DbCache
Take in mind possibility of data loss with this option.

**Comment by chanfr** [comment]
I'm having issues using request_cache with threads with the default sqlite backend.

I'm making requests that take a short time to finish with 40 threads and in some cases, I'm getting the following error:

database is locked


I'm using Pool from multipressing

from multiprocessing import Pool


    pool = Pool(n_cores)
    df = pd.concat(pool.map(func, df_split))


where func performs a get request.

**Comment by ximo1984** [comment]
@chanfr , I have this error too, could you fix it? Thanks in advance.

**Comment by JWCook** [comment]
@chanfr @ximo1984 This is likely fixed in the current master branch with PR #85. Test that out if you get the chance, and please let me know if this is still an issue.

**Comment by ximo1984** [comment]
@JWCook Sorry but I'm not working on that project currently so I can't answer.
